nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -1517,11 +1534,11 @@ trait NirGenExpr { self: NirGenPhase =>
       val Seq(lengthp) = argsp
       val length = genExpr(lengthp)
 
-      buf.arrayalloc(genType(targ), length, unwind)
+      buf.arrayalloc(genType(targ), length, unwind)(targ.sym.pos)
```

Comment:
That would be the position of the declaration of the element type. Nothing to do with the position of the `new Array`. You need to use the `pos` of the `Tree` that caused the call to `genApplyNewArray`.

In general, you can never use the position of a `Symbol`. You must always use the position of a `Tree`.